### PR TITLE
Golang & Docker workflows: add db caching to trivy

### PR DESCRIPTION
**Pull request by chicco785**
## Description

Cache trivy db daily to avoid incurring in the TOO MANY REQUEST issue.

## Changes Made

* Cache trivy db daily in golang wf
* Cache trivy db daily in docker wf

## Related Issues

N/A

## Checklist

- [x] I have used a PR title that is descriptive enough for a release note.
- [x] I have tested these changes locally.
- [ ] I have added appropriate tests or updated existing tests.
- [x] I have tested these changes on storer
- [ ] I have added appropriate documentation or updated existing documentation.

### CODE REVIEW

**Comment by chicco785** [comment]
it was tested here:
https://github.com/zaphiro-technologies/storer/actions/runs/11178401264/job/31076141774

File: .github/workflows/docker.yaml [hunk 1]
```diff
@@ -143,6 +143,17 @@ jobs:
...
+      - name: Get data
+        id: date
+        run: echo "date=$(date +%Y-%m-%d)" >> $GITHUB_OUTPUT
+      - name: Restore trivy cache
+        uses: actions/cache@v4
+        with:
+          path: cache/db
+          key: trivy-cache-${{ steps.date.outputs.date }}
+          restore-keys:
+            trivy-cache-
```
Comment (hiimjako) [inline .github/workflows/docker.yaml:156]:
```suggestion
            trivy-cache
```
Is this a typo?

File: .github/workflows/golang.yaml [hunk 2]
```diff
@@ -246,6 +246,17 @@ jobs:
...
+      - name: Get data
+        id: date
+        run: echo "date=$(date +%Y-%m-%d)" >> $GITHUB_OUTPUT
+      - name: Restore trivy cache
+        uses: actions/cache@v4
+        with:
+          path: cache/db
+          key: trivy-cache-${{ steps.date.outputs.date }}
+          restore-keys:
+            trivy-cache-
```
Comment (hiimjako) [inline .github/workflows/golang.yaml:259]:
```suggestion
            trivy-cache
```
Same here

**Review by hiimjako — APPROVED** [review]
Is the cache for the entire repository or just the individual branch?

**Comment by chicco785** [comment]
> Is the cache for the entire repository or just the individual branch?

all repository, since the key is the date ;)

**Pull request merged**